File: frontend/src/types/mission.ts
export type Waypoint = [number, number]; // [longitude, latitude]

export interface MissionRoute {
  id: number;
  name: string;
  description: string;
  waypoints: Waypoint[];
  created_by: string;
  created_at: string;
  updated_at: string;
}

export interface MissionRouteCreate {
  name: string;
  description: string;
  waypoints: Waypoint[];
}

File: backend/app/models/mission.py
from pydantic import BaseModel
from typing import List, Optional, Tuple
from datetime import datetime

Waypoint = Tuple[float, float]  # (longitude, latitude)


class MissionRouteCreate(BaseModel):
    name: str
    description: Optional[str] = ""
    waypoints: List[Waypoint]


class MissionRouteOut(BaseModel):
    id: int
    name: str
    description: Optional[str]
    waypoints: List[Waypoint]
    created_by: str
    created_at: datetime
    updated_at: datetime


class DispatchOrderRequest(BaseModel):
    waypoints: List[Waypoint]
